benchmarks/benchmarks/common.py

```diff
@@ -25,40 +25,90 @@
     'complex256',
 ]
 
+
+def memoize(func):
+    result = []
+    def wrapper(*a, **kw):
+        if not result:
+            result.append(func(*a, **kw))
+        return result[0]
+    return wrapper
```

Comment:
My little sample code was intentional about not allowing `args` and `kwargs`, because they will be ignored on subsequent calls in any case. (Fancier memoization systems use them as keys into a dict of results, and then have to worry about making them hashable, blah blah blah... but in this case there's no point in bothering with all that.)

Not that it matters terribly much for this little out-of-the-way utility routine :-) I'd probably remove the args/kwargs handling, but eh. Feel free to merge when you think it's ready.